src/bluehawk/processor/Processor.ts

```diff
@@ -23,16 +23,22 @@ export type Listener = (result: ParseResult) => void | Promise<void>;
 
 export class Processor {
   processors: CommandProcessors = {};
-  listeners: Listener[] = [];
+  listeners = new Set<Listener>();
 
   // Subscribe to processed file events
   subscribe(listener: Listener): void {
-    this.listeners.push(listener);
+    if (this.listeners.has(listener)) {
```

Comment:
I like the changes here to move towards a set, but what are the conditions that a new listener overwrite an old one and that should be rejected?

Reply:
Yeah, we don't need this warning. The CLI might re-add its listeners as a matter of course when getting the Bluehawk instance.